File: android/src/main/kotlin/com/usercentrics/sdk/flutter/bridge/RestoreUserSessionBridge.kt
package com.usercentrics.sdk.flutter.bridge

import com.usercentrics.sdk.flutter.api.FlutterMethodCall
import com.usercentrics.sdk.flutter.api.FlutterResult
import com.usercentrics.sdk.flutter.api.UsercentricsProxy
import com.usercentrics.sdk.flutter.api.UsercentricsProxySingleton
import com.usercentrics.sdk.flutter.serializer.serialize

internal class RestoreUserSessionBridge(
    private val usercentrics: UsercentricsProxy = UsercentricsProxySingleton
) : MethodBridge {

    companion object {
        private const val restoreUserSessionErrorCode =
            "usercentrics_flutter_restoreUserSession_error"
    }

    override val name: String
        get() = "restoreUserSession"

    override fun invoke(call: FlutterMethodCall, result: FlutterResult) {
        assert(name == call.method)
        usercentrics.instance.restoreUserSession(
            controllerId = call.arguments as String,
            onSuccess = {
                result.success(it.serialize())
            },
            onFailure = {
                result.error(
                    restoreUserSessionErrorCode,
                    it.message,
                    it
                )
            }
        )
    }

}

File: ios/Classes/Bridge/RestoreUserSessionBridge.swift
import Usercentrics
import Flutter

struct RestoreUserSessionBridge : MethodBridge {

    let name: String = "restoreUserSession"
    let usercentrics: UsercentricsProxyProtocol

    func invoke(_ call: FlutterMethodCall, _ result: @escaping FlutterResult) {
        assert(call.method == name)
        guard let argument = call.arguments as? String else {
            result(FlutterError(code: "usercentrics_flutter_restoreUserSession_error",
                                message: "Invalid Parameter",
                                details: nil ))
            return
        }

        usercentrics.shared.restoreUserSession(controllerId: argument) { status in
            result(status.serialize())
        } onFailure: { error in
            result(FlutterError(code: "usercentrics_flutter_restoreUserSession_error",
                                message: error.localizedDescription,
                                details: nil ))
        }
    }
}
